### PR TITLE
Makefile - instead of calling `as` directly, use it from env var

### PR DESCRIPTION
**Pull request by rahilarious**
#### What type of PR is this?

> /kind api-change

/kind bug

> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Which issue(s) this PR fixes:

> /bin/sh: line 1: as: command not found`

error where `as` symlink doesn't exists

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM
A couple of tests are spitting up, I think it's unrelated and have restarted them.

**Review by edsantiago — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,7 @@ BUILDFLAGS := -tags "$(BUILDTAGS)"
 BUILDAH := buildah
 SELINUXOPT ?= $(shell test -x /usr/sbin/selinuxenabled && selinuxenabled && echo -Z)
 SELINUXTYPE=container_runtime_exec_t
+AS ?= as
```
Comment (nalind) [inline Makefile:19]: Is there a platform we build on where this isn't provided by a default rule, as `LD` is?

Comment (rahilarious) [reply Makefile:19]: This is for the case where `as` command doesn't exist. Instead `x86-64-gnu-pc-linux-as` (or something similar) should be called directly.

Comment (rahilarious) [reply Makefile:19]: Honestly, you can drop from HEAD
```
AS ?= as
STRIP ?= strip
```
I added them to make sure to not break something accidentally

Comment (nalind) [reply Makefile:19]: `make -p -f /dev/null` with GNU make 4.4.1 on Fedora isn't showing me a predefined value for STRIP, so we'd need to keep that.

**Pull request merged**